examples/preprocessing/plot_find_ref_artifacts.py

```diff
@@ -0,0 +1,126 @@
+"""
+.. _ex-megnoise_processing:
+
+====================================
+Find MEG reference channel artefacts
+====================================
+
+Use ICA decompositions of MEG reference channels to remove intermittent noise.
+
+Many MEG systems have an array of reference channels which are used to remove
+external magnetic noise. However, standard removal techniques often fail when
+noise is intermittent. This technique often succeeds where the standard
+techniques do not.
+
+There are two algorithms to choose from: separate and together (default). In
+the "separate" algorithm, two ICA decompositions are made: one on the reference
+channels, and one on reference + standard channels. The reference + standard
+channel components which correlate with the reference channel components are
+removed.
+
+In the "together" algorithm, a single ICA decomposition is made on reference +
+standard channels, and those components whose weights are particularly heavy
+on the reference channels are removed.
+
+This technique is fully described and validated in :footcite:`HannaEtAl2020`
+
+"""
+# Authors: Jeff Hanna <jeff.hanna@gmail.com>
+#
+# License: BSD (3-clause)
+
+import mne
+from mne import io
+from mne.datasets import refmeg_noise
+from mne.preprocessing import ICA
+
+print(__doc__)
+
+data_path = refmeg_noise.data_path()
+
+###############################################################################
+# Read raw data
+raw_fname = data_path + '/sample_reference_MEG_noise-raw.fif'
+raw = io.read_raw_fif(raw_fname, preload=True)
+
+###############################################################################
+# Note that even though standard noise removal has already
+# been applied to these data, much of the noise in the reference channels
+# (bottom of the plot) can still be seen in the standard channels.
+raw.plot(duration=800, n_channels=160, scalings={"mag": 8e-13,
+                                                 "ref_meg": 2e-11})
+
+###############################################################################
+# The PSD of these data show the noise as clear peaks
+raw.plot_psd(fmax=30)
+
+###############################################################################
+# Run the "together" algorithm
+raw_tog = raw.copy()
+ica_kwargs = dict(
+    method='picard',
+    fit_params=dict(tol=1e-3),  # use a high tol here for speed
+)
+all_picks = mne.pick_types(raw_tog.info, meg=True, ref_meg=True)
+ica_tog = ICA(n_components=60, allow_ref_meg=True, **ica_kwargs)
+ica_tog.fit(raw_tog, picks=all_picks)
+bad_comps, scores = ica_tog.find_bads_ref(raw_tog, threshold=2.5)
+
+# Plot scores with bad components marked
+ica_tog.plot_scores(scores, bad_comps)
```

Comment:
this plot looks unusual to me; why is the *smallest* bar the one that is highlighted?  Is this a feature of using ref_meg channels in the ICA?  I'm used to seeing the *largest* bar be the one marked as "bad", like [here](https://mne.tools/dev/auto_tutorials/preprocessing/plot_40_artifact_correction_ica.html#using-an-eog-channel-to-select-ica-components)